### PR TITLE
HV: minor refine about coding style

### PR DESCRIPTION
**Pull request by lifeix**
minor refine about coding style

Tracked-On: #861
Signed-off-by: Li, Fei1 <fei1.li@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
2002<br>No new violations to the coding guideline detected.

File: hypervisor/lib/string.c [hunk 1]
```diff
@@ -439,28 +439,26 @@ char *strstr_s(const char *str1, size_t maxlen1, const char *str2, size_t maxlen
...
-			if (str1[i] != str2[i]) {
+			if (str1[i + j] != str2[i]) {
 				break;
 			}
 		}
 		if (i == len2) {
 			return (char *)str1;
 		}
-		str1++;
+		j++;
```
Comment (anthonyzxu) [inline hypervisor/lib/string.c]: add a local variable for str1

Comment (lifeix) [reply hypervisor/lib/string.c]: Thanks for pointing out the bug. Fixed it.

File: hypervisor/lib/string.c [hunk 2]
```diff
@@ -439,28 +440,26 @@ char *strstr_s(const char *str1, size_t maxlen1, const char *str2, size_t maxlen
...
+	/* return str1 if str2 equals to str1 or str2 points to a string with zero length*/
+	if ((str1 == str2) || (len2 == 0U)) {
 		return (char *)str1;
 	}
 
+	str = str1;
 	while (len1 >= len2) {
 		for (i = 0U; i < len2; i++) {
-			if (str1[i] != str2[i]) {
+			if (str[i] != str2[i]) {
```
Comment (dongyaozu) [inline hypervisor/lib/string.c]: we already have str1/str2, adding "str" is ugly... how about pstr, or str0 ?

File: hypervisor/arch/x86/vmx.c [hunk 3]
```diff
@@ -408,6 +409,8 @@ void vmx_write_cr0(struct acrn_vcpu *vcpu, uint64_t cr0)
 				vcpu_get_efer(vcpu) & ~MSR_IA32_EFER_LMA_BIT);
 		} else {
 		}
```
Comment (dongyaozu) [inline hypervisor/arch/x86/vmx.c]: This else {}  seems to be not necessary

File: hypervisor/arch/x86/vmx.c [hunk 4]
```diff
@@ -408,6 +409,8 @@ void vmx_write_cr0(struct acrn_vcpu *vcpu, uint64_t cr0)
...
+	} else {
+		/* do nothing */
```
Comment (dongyaozu) [inline hypervisor/arch/x86/vmx.c]: We need this one here.

File: hypervisor/Makefile [hunk 5]
```diff
@@ -357,7 +359,8 @@ $(VERSION):
 	echo "#define HV_BUILD_VERSION "\""$$PATCH"\""" >> $(VERSION);\
 	echo "#define HV_BUILD_TYPE "\""$$BUILD_TYPE"\""" >> $(VERSION);\
 	echo "#define HV_BUILD_TIME "\""$$TIME"\""" >> $(VERSION);\
-	echo "#define HV_BUILD_USER "\""$$USER"\""" >> $(VERSION)
+	echo "#define HV_BUILD_USER "\""$$USER"\""" >> $(VERSION);\
+	echo "#endif" >> $(VERSION)
```
Comment (dongyaozu) [inline hypervisor/Makefile]: @binbinwu1  can you review this patch?

File: hypervisor/arch/x86/vmx.c (hunk 3 above)
Comment (lifeix) [reply hypervisor/arch/x86/vmx.c]: Yes. It's not necessary.

File: hypervisor/lib/string.c (hunk 2 above)
Comment (lifeix) [reply hypervisor/lib/string.c]: str1 -> haystack
str2 -> needle ?

Comment (dongyaozu) [reply hypervisor/lib/string.c]: What does this mean? too complicated?

File: hypervisor/arch/x86/vmx.c (hunk 3 above)
Comment (dongyaozu) [reply hypervisor/arch/x86/vmx.c]: delete it?

Comment (lifeix) [reply hypervisor/arch/x86/vmx.c]: done

**Comment by HuiHuangShi** [comment]
2002<br>1 new potential violations to the coding guideline detected. Please help check. Thanks!  <br><br>1.filename:/hypervisor/lib/string.c&nbsp;&nbsp;function:strstr_s&nbsp;&nbsp;offset:460:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Cast on a constant value.                    : pstr<br><br>

**Pull request merged**